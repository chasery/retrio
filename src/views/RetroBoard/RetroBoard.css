.RetroBoard {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  max-width: 100%;
  padding: 0;
}

.RetroBoard__board {
  display: grid;
  flex-grow: 1;
  grid-template-columns: 1fr;
  grid-template-rows: repeat(4, 50%);
}

@media all and (min-width: 30rem) {
}

@media all and (min-width: 50rem) {
  .RetroBoard__board {
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: repeat(2, 50%);
  }
}

@media all and (min-width: 64rem) {
  .RetroBoard__board {
    grid-template-columns: repeat(4, 1fr);
    grid-template-rows: repeat(1, 100%);
  }
}
