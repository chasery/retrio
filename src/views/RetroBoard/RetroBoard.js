import React, { useContext, useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import RetrioContext from '../../context/retrio-context';
import BoardsApiService from '../../services/boards-api-service';
import Header from '../../components/Header/Header';
import RetroBoardColumn from '../../components/RetroBoardColumn/RetroBoardColumn';
import Error from '../../components/Error/Error';
import './RetroBoard.css';

function RetroBoard(props) {
  const context = useContext(RetrioContext);
  const { boardId } = useParams();
  const [board, setBoard] = useState({ cards: [] });
  const [error, setError] = useState(null);

  useEffect(() => {
    async function initState() {
      try {
        let apiCall = await BoardsApiService.getBoard(boardId);
        let res = await apiCall;
        setBoard(res);
      } catch (error) {
        setError(error.error);
      }
    }

    initState();
  }, [boardId]);

  const renderColumns = (cards) => {
    const columns = [];

    for (const [key, value] of Object.entries(context.cardCategories)) {
      columns.push(
        <RetroBoardColumn
          key={key}
          boardOwner={board.owner}
          category={key}
          title={value}
          cards={
            cards.length !== 0
              ? cards.filter((card) => card.category === parseFloat(key))
              : []
          }
        />
      );
    }

    return columns;
  };

  return (
    <>
      <Header fullWidth={true} />
      <main role='main'>
        <section className='RetroBoard'>
          {error ? (
            <Error message={error} />
          ) : (
            <ul className='RetroBoard__board'>{renderColumns(board.cards)}</ul>
          )}
        </section>
      </main>
    </>
  );
}

export default RetroBoard;
