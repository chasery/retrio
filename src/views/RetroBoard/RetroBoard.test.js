import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RetroBoard from './RetroBoard';

describe('RetroBoard component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<RetroBoard />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
