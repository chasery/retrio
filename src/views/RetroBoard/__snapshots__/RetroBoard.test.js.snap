// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetroBoard component renders without error 1`] = `
<Fragment>
  <Header
    fullWidth={true}
  />
  <main
    role="main"
  >
    <section
      className="RetroBoard"
    >
      <ul
        className="RetroBoard__board"
      >
        <RetroBoardColumn
          boardId={1}
          cards={
            Array [
              Object {
                "headline": "We created an API to add a card!",
                "id": 1,
                "submitted_by": "Ryan Chase",
                "text": "It’s crazy, we actually made a restful API endpoint to add a note to the retro board. Thanks Bob for helping drive the discussions with our PO around the buisness logic for this. We couldn’t have done this with out you!",
              },
              Object {
                "headline": "We created an API to add a card!",
                "id": 2,
                "submitted_by": "Ryan Chase",
                "text": "It’s crazy, we actually made a restful API endpoint to add a note to the retro board. Thanks Bob for helping drive the discussions with our PO around the buisness logic for this. We couldn’t have done this with out you!",
              },
              Object {
                "headline": "We created an API to add a card!",
                "id": 3,
                "submitted_by": "Ryan Chase",
                "text": "It’s crazy, we actually made a restful API endpoint to add a note to the retro board. Thanks Bob for helping drive the discussions with our PO around the buisness logic for this. We couldn’t have done this with out you!",
              },
            ]
          }
          title="What went well"
        />
        <RetroBoardColumn
          boardId={1}
          cards={
            Array [
              Object {
                "headline": "We created an API to add a card!",
                "id": 4,
                "submitted_by": "Ryan Chase",
                "text": "It’s crazy, we actually made a restful API endpoint to add a note to the retro board. Thanks Bob for helping drive the discussions with our PO around the buisness logic for this. We couldn’t have done this with out you!",
              },
              Object {
                "headline": "We created an API to add a card!",
                "id": 5,
                "submitted_by": "Ryan Chase",
                "text": "It’s crazy, we actually made a restful API endpoint to add a note to the retro board. Thanks Bob for helping drive the discussions with our PO around the buisness logic for this. We couldn’t have done this with out you!",
              },
            ]
          }
          title="What didn't go well"
        />
        <RetroBoardColumn
          boardId={1}
          cards={
            Array [
              Object {
                "headline": "We created an API to add a card!",
                "id": 6,
                "submitted_by": "Ryan Chase",
                "text": "It’s crazy, we actually made a restful API endpoint to add a note to the retro board. Thanks Bob for helping drive the discussions with our PO around the buisness logic for this. We couldn’t have done this with out you!",
              },
            ]
          }
          title="To try"
        />
        <RetroBoardColumn
          boardId={1}
          cards={
            Array [
              Object {
                "headline": "We created an API to add a card!",
                "id": 7,
                "submitted_by": "Ryan Chase",
                "text": "It’s crazy, we actually made a restful API endpoint to add a note to the retro board. Thanks Bob for helping drive the discussions with our PO around the buisness logic for this. We couldn’t have done this with out you!",
              },
              Object {
                "headline": "We created an API to add a card!",
                "id": 8,
                "submitted_by": "Ryan Chase",
                "text": "It’s crazy, we actually made a restful API endpoint to add a note to the retro board. Thanks Bob for helping drive the discussions with our PO around the buisness logic for this. We couldn’t have done this with out you!",
              },
            ]
          }
          title="Shout outs"
        />
      </ul>
    </section>
  </main>
</Fragment>
`;
